tests/unit_tests/test_map_pool.py

```diff
@@ -48,6 +88,24 @@ def test_choose_map_all_maps_played(map_pool_factory):
     assert chosen_map in maps
 
 
+def test_choose_map_all_played_but_generated_map_doesnt_dominate(map_pool_factory):
+    maps = [
+        Map(1, "some_map", "maps/some_map.v001.zip", 1000000),
```

Comment:
why do theses have big weight?

Reply:
This is to ensure that they are chosen as one of them should be chosen unless they are not included which would happen if the check for Map type in the least common count didnt happen.

Reply:
Right, ok